asteroidsyncserviced/bluez/bluezclient.cpp

```diff
@@ -100,13 +109,17 @@ BluezClient::BluezClient(QObject *parent):
                 }
             }
         }
+    } 
+    else 
+    {
+        throw std::runtime_error("unable to connect to D-Bus");
```

Comment:
The above `std::bad_alloc` discussion is actually a very good introduction to why I believe you should just completely avoid using exceptions. https://pianomanfrazier.com/post/exceptions-considered-harmful/

Notice that the Qt framework does not use any exception ever, precisely because it's a horrible language construct.

Also here you're throwing an exception but never catching it so instead of gracefully continuing the execution this will cause a hard crash. I recommend you just log something to the standard error output instead.

Reply:
This is now changed to use qCritical to emit a message instead.